src/main/java/frc/robot/undertaker/SimUndertaker.java

```diff
@@ -1,61 +1,50 @@
 package frc.robot.undertaker;
 
-import com.revrobotics.REVPhysicsSim;
-import com.revrobotics.CANSparkMax;
-import com.revrobotics.CANSparkLowLevel.MotorType;
-import edu.wpi.first.math.system.plant.DCMotor;
-import frc.robot.Constants;
-
 public class SimUndertaker implements IUndertaker {
 
-  private final CANSparkMax intakeMotorOne = new CANSparkMax(Constants.Undertaker.INTAKE_MOTOR_ONE, MotorType.kBrushless);
-  private final CANSparkMax intakeMotorTwo = new CANSparkMax(Constants.Undertaker.INTAKE_MOTOR_TWO, MotorType.kBrushless);
-  private boolean[] mockMotorStatus = { false, false }; 
-  private double desiredSpeed;
-  private boolean hasRun = false;
-
-  public SimUndertaker(){
-    REVPhysicsSim.getInstance().addSparkMax(intakeMotorOne, DCMotor.getNEO(1));
-    REVPhysicsSim.getInstance().addSparkMax(intakeMotorTwo, DCMotor.getNEO(1));
-  }
-
-  @Override
-  public double getDesiredSpeed(){
-    return desiredSpeed;
-  }
-
-  @Override
-  public double[] getMotorsOutput(){
-    double[] motorsOutput = new double[2];
-    motorsOutput[0] = intakeMotorOne.getAppliedOutput();
-    motorsOutput[1] = intakeMotorTwo.getAppliedOutput();
-    return motorsOutput;
-  }
-
-  @Override
-  public void run(double speed){
-    if (mockMotorStatus[0] && mockMotorStatus[1]) {
-      this.desiredSpeed = speed;
-      intakeMotorOne.setVoltage(speed * 12.0);
-      intakeMotorTwo.setVoltage(speed * 12.0);
-      hasRun = true;
+    private boolean[] mockMotorStatus = { false, false };
+    private double desiredSpeed;
+    private boolean hasRun = false;
```

Comment:
This variable (and the code in the undertaker unit tests) tell me that this code is doing 2 things when it should really be doing one. This implementation of IUndertaker should only be making an implementation for using and representing the undertaker state in the sim GUI.

There should be a new implementation called `TestUndertaker` that stores the variables and provides the methods needed to support the unit test suite (`setIsAlive`, `getHasRun`, `resetHasRun`) that tests the undertaker subsystem.

Reply:
attempted to fix